### PR TITLE
Wasm ci updates

### PR DESCRIPTION
**Pull request by kquick**
Mostly this is just CI stuff, with a few code tweaks for older GHC (8.6.5) for compatibility with the rest of crucible.

### CODE REVIEW

File: .github/workflows/crux-llvm-build.yml [hunk 1]
```diff
@@ -1,4 +1,10 @@
 name: crux-llvm
+
+# Note: also tests crucible-wasm builds, which aren't technically
+# crux-llvm, but since crucible-wasm depends on crucible and
+# crucible-llvm, it was easier to add it here than to creati a
```
Comment (RyanGlScott) [inline .github/workflows/crux-llvm-build.yml:5]:
```suggestion
# crucible-llvm, it was easier to add it here than to create a
```

File: .github/workflows/crux-llvm-build.yml [hunk 2]
```diff
@@ -186,9 +192,16 @@ jobs:
...
       - shell: bash
+        name: Build crux-llvm SVComp tool
         run: .github/ci.sh build exe:crux-llvm-svcomp
         if: runner.os != 'Windows'
 
+      - shell: bash
+        name: Build crucible wasm support library
+        run: |
+          (cd dependencies/haskell-wasm; nix-shell -p hpack --run hpack)
+          ./github/ci.sh build crucible-wasm
```
Comment (RyanGlScott) [inline .github/workflows/crux-llvm-build.yml:203]:
```suggestion
          ./.github/ci.sh build crucible-wasm
```

I think?

File: .github/workflows/crux-llvm-build.yml (hunk 1 above)
Comment (kquick) [reply .github/workflows/crux-llvm-build.yml]: oops

File: .github/workflows/crux-llvm-build.yml (hunk 2 above)
Comment (kquick) [reply .github/workflows/crux-llvm-build.yml]: Actually, I just need to remove the slash, but good catch.  The moral of the story is one shouldn't code while distracted.

**Pull request merged**